From e4ba70b64a56dd24976c5b14a9a1e2a8a2358c67 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 3 Sep 2014 15:15:49 -0400
Subject: [ethernet] bna: bnad code cleanup

Message-id: <1409757361-26781-2-git-send-email-ivecera@redhat.com>
Patchwork-id: 91162
O-Subject: [RHEL7.1 PATCH 01/13] bnad: code cleanup
Bugzilla: 1136912
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

BZ: #1136912

Upstream commit(s):
commit 2fd888a5e7b8fc1f1d496fa5379103d667852859
Author: stephen hemminger <stephen@networkplumber.org>
Date:   Wed Jan 15 08:24:21 2014 -0800

    bnad: code cleanup

    Use 'make namespacecheck' to code that could be declared static.
    After that remove code that is not being used.

    Compile tested only.

    Signed-off-by: Stephen Hemminger <stephen@networkplumber.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/brocade/bna/bfa_ioc.c b/drivers/net/ethernet/brocade/bna/bfa_ioc.c
index 537bba1..1803c39 100644
--- a/drivers/net/ethernet/brocade/bna/bfa_ioc.c
+++ b/drivers/net/ethernet/brocade/bna/bfa_ioc.c
@@ -1147,25 +1147,6 @@ bfa_nw_ioc_sem_release(void __iomem *sem_reg)
  writel(1, sem_reg);
 }
 
-/* Invalidate fwver signature */
-enum bfa_status
-bfa_nw_ioc_fwsig_invalidate(struct bfa_ioc *ioc)
-{
- u32 pgnum, pgoff;
- u32 loff = 0;
- enum bfi_ioc_state ioc_fwstate;
-
- ioc_fwstate = bfa_ioc_get_cur_ioc_fwstate(ioc);
- if (!bfa_ioc_state_disabled(ioc_fwstate))
-  return BFA_STATUS_ADAPTER_ENABLED;
-
- pgnum = bfa_ioc_smem_pgnum(ioc, loff);
- pgoff = PSS_SMEM_PGOFF(loff);
- writel(pgnum, ioc->ioc_regs.host_page_num_fn);
- writel(BFI_IOC_FW_INV_SIGN, ioc->ioc_regs.smem_page_start + loff);
- return BFA_STATUS_OK;
-}
-
 /* Clear fwver hdr */
 static void
 bfa_ioc_fwver_clear(struct bfa_ioc *ioc)
@@ -1780,15 +1761,9 @@ bfa_flash_raw_read(void __iomem *pci_bar, u32 offset, char *buf,
  return BFA_STATUS_OK;
 }
 
-u32
-bfa_nw_ioc_flash_img_get_size(struct bfa_ioc *ioc)
-{
- return BFI_FLASH_IMAGE_SZ/sizeof(u32);
-}
-
 #define BFA_FLASH_PART_FWIMG_ADDR 0x100000 /* fw image address */
 
-enum bfa_status
+static enum bfa_status
 bfa_nw_ioc_flash_img_get_chnk(struct bfa_ioc *ioc, u32 off,
          u32 *fwimg)
 {
diff --git a/drivers/net/ethernet/brocade/bna/bnad.c b/drivers/net/ethernet/brocade/bna/bnad.c
index c892d6f..4ad1187 100644
--- a/drivers/net/ethernet/brocade/bna/bnad.c
+++ b/drivers/net/ethernet/brocade/bna/bnad.c
@@ -2111,7 +2111,7 @@ bnad_rx_ctrl_init(struct bnad *bnad, u32 rx_id)
 }
 
 /* Called with mutex_lock(&bnad->conf_mutex) held */
-u32
+static u32
 bnad_reinit_rx(struct bnad *bnad)
 {
  struct net_device *netdev = bnad->netdev;
-- 
1.7.1